pkg/binding/definition.go

```diff
@@ -140,25 +141,24 @@ type mapFromDataFieldDefinition struct {
 	objectType            objectType
 	outputName            string
 	sourceValue           string
-	path                  []string
+	definition
 }
 
 var _ Definition = (*mapFromDataFieldDefinition)(nil)
 
-func (d *mapFromDataFieldDefinition) GetPath() []string { return d.path }
-
 func (d *mapFromDataFieldDefinition) Apply(u *unstructured.Unstructured) (Value, error) {
 	if d.secretConfigMapReader == nil {
 		return nil, errors.New("kubeClient required for this functionality")
 	}
 
-	resourceName, ok, err := unstructured.NestedString(u.Object, d.path...)
+	res, err := getValuesByJSONPath(u.Object, d.path)
 	if err != nil {
 		return nil, err
 	}
-	if !ok {
-		return nil, errors.New("not found")
+	if len(res) != 1 {
+		return nil, fmt.Errorf("Only one value should be returned for %v but we got %v", d.path, res)
```

Comment:
Error strings should not be capitalized.
Ref. https://github.com/golang/go/wiki/CodeReviewComments#error-strings